common/shared.c

```diff
@@ -460,3 +464,50 @@ void ft_parsecsopts(int op, char *optarg, struct cs_opts *opts)
 	}
 }
 
+void ft_integ_fill_buf(void *buf, int size, int iter)
+{
+	char *msg_buf;
+	int msg_index;
+	int j;
+
+	msg_index = (iter*INTEG_SEED) % integ_alphabet_length;
+	msg_buf = (char *)buf;
+	for (j = 0; j < size; j++) {
+		msg_buf[j] = integ_alphabet[msg_index++];
+		if (msg_index >= integ_alphabet_length) {
+			msg_index = 0;
+		}
+	}
+}
+
+int ft_integ_check_buf(void *buf, int size, int iter)
+{
+	char *recv_data;
+	char c;
+	int msg_index;
+	int j;
+	int first_error_byte;
+
+	msg_index = (iter*INTEG_SEED) % integ_alphabet_length;
+	recv_data = (char *)buf;
+	first_error_byte = -1;
+
+	for(j = 0; j < size; j++) {
+		c = integ_alphabet[msg_index++];
+		if (msg_index >= integ_alphabet_length) {
+			msg_index = 0;
+		}
+		if (c == recv_data[j]) {
+			continue;
+		}
+		first_error_byte = j;
+		break;
+	}
```

Comment:
The structure of this for loop is odd.  Please change to:
if (c != recv_data[j])
   break;